cmd/monitor/config.go

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"github.com/sigstore/rekor-monitor/pkg/identity"
+	"github.com/sigstore/rekor-monitor/pkg/notifications"
+)
+
+type IdentityMonitorConfiguration struct {
```

Comment:
Should we also include `once` so this either runs indefinitely or only for one iteration? 

If so, this would need to be restructured a bit, so that we continually pull in the start index from the last checkpoint that was recorded.

Reply:
related, I think if you specify start/end, that should effectively be the same as `once`, as a one-off search across indices (which could be used if you are running the search for the first time, for example, or catching up)

Reply:
I was considering how to include once but I think that makes sense for constantly pulling in the last checkpoint, will do

Reply:
We should also decide between the indefinite loop being in the cmd package or in the inner API. Right now, the functionality is embedded in the API, and I wouldn’t expect API users to need to loop indefinitely. Also it’s not possible to test an endless loop, so we can maintain good coverage in the API where the cmd package isn’t expected to have coverage.

Reply:
looping in the cmd package definitely makes more sense- will implement that way in this PR, and refactor consistency check as such in #492